### DISCUSSION TITLE
`nb::ndarray`, constraint on `nb::ndim<1>` or `nb::ndim<2>`

### DISCUSSION THREAD

**Discussion by francesco-ballarin** [wjakob/nanobind#354] [Q&A] [answered]
Hi,
I have a function than can take as input either a `nb::ndarray<const std::int32_t, nb::ndim<1>, nb::c_contig>` or a `nb::ndarray<const std::int32_t, nb::ndim<2>, nb::c_contig>` (note `nb::ndim<1>` vs `nb::ndim<2>`).
Is there any way I can constrain the shape, but avoiding to write the wrapper twice?
I guess I am looking for something like `nb::ndarray<const std::int32_t, nb::ndim<1> || nb::ndim<2>, nb::c_contig>`.

Thanks,
Francesco

**Comment by wjakob** [comment, marked as answer]
You could not have a dimension annotation at all and deal with it at runtime. The ndarray docs show how this can be done with array views. Otherwise, the ``std::variant`` type caster may come in handy.
